app/gui2/src/util/toast.ts

```diff
@@ -25,6 +26,14 @@ export function useToast(options: UseToastOptions = {}) {
       if (toast.isActive(id)) toast.update(id, { ...options, render: content })
       else toast(content, { ...options, toastId: id })
     },
+    reportError<T, E>(result: Result<T, E>, preamble?: string) {
+      if (!result.ok) {
+        const msg = result.error.message(preamble)
+        console.error(msg)
+        this.show(msg)
+      }
+      return result
```

Comment:
Not sure about this. Putting the error check condition into this function forces the `preamble` string to be always computed, and it is sort of awkward to use when you have to check for `ok` everywhere anyway. I think this should rather receive a `ResultError` as argument and just handle printing it. We can always combine this with a function like `mapError` on results.